src/app/core/services/notifications/message-handler.service.ts

```diff
@@ -20,11 +23,15 @@ export class MessageHandlerService {
     public usage: UsageService,
     public appConfig: AppConfigService
   ) {
-    this.firebase
+    this.messageListener = this.firebase
       .onMessageReceived()
       .subscribe(data => this.onMessageReceived(new Map(Object.entries(data))))
   }
 
+  ngOnDestroy() {
+    this.messageListener.unsubscribe()
+  }
```

Comment:
what happens here when the app is closed? The subscription will be unsubscribed? What if we receive an FCM message then, it will not trigger this `onMessagedReceived` callback?

Reply:
The data message will be received on the Firebase side, but since we are only able to execute the callback when the app is opened or in-background, it will still behave the same (subscribing when the app is open).